compiler/src/iree/compiler/Dialect/Flow/Transforms/DetachElementwiseFromNamedOps.cpp

```diff
@@ -111,6 +112,62 @@ struct DetachElementwisePattern
   }
 };
 
+/// Replace uses of splat constants as `outs` operands of `LinalgExt`
+/// operations. More canonical representation is to use a `init_tensor -> fill
+/// -> outs` operand sequence. Splat constants pulled in this way causes issues
+/// with allocations. Using `fill` will allow for fusing with the op just like
+/// fill -> linalg ops are fused. If not as a fallback they would be converted
+/// to a splat, but both without stack allocations.
+struct DetachSplatConstantOutsOperands
+    : public OpInterfaceRewritePattern<IREE::LinalgExt::LinalgExtOp> {
+  using OpInterfaceRewritePattern<
+      IREE::LinalgExt::LinalgExtOp>::OpInterfaceRewritePattern;
+
+  LogicalResult matchAndRewrite(IREE::LinalgExt::LinalgExtOp linalgExtOp,
+                                PatternRewriter &rewriter) const {
+    SmallVector<Value> newOutsOperands;
+    bool madeChanges = false;
+    for (auto outsOperandNum :
+         llvm::seq<unsigned>(0, linalgExtOp.getNumOutputs())) {
+      OpOperand *outOperand = linalgExtOp.getOutputOperand(outsOperandNum);
+      auto constOp = outOperand->get().getDefiningOp<arith::ConstantOp>();
+      if (!constOp) continue;
+
+      auto resultType =
+          constOp.getResult().getType().dyn_cast<RankedTensorType>();
+      if (!resultType || !resultType.getElementType().isIntOrFloat()) continue;
+
+      auto attr = constOp.getValue().dyn_cast<DenseElementsAttr>();
+      if (!attr || !attr.isSplat()) continue;
+
+      Location loc = constOp.getLoc();
+      Type elementType = resultType.getElementType();
+      Value initTensorOp = rewriter.create<linalg::InitTensorOp>(
+          loc, resultType.getShape(), elementType);
+      Attribute constValue;
+      if (elementType.isa<IntegerType>()) {
```

Comment:
I think stella was playing with tensor<index> types that we may want to handle here as well - or at least fail the pattern if hit (vs going down the float path)

Reply:
Above I check the element type is `intOrFloat`...